cpp/src/cider/tests/CiderNewHashTableTest.cpp

```diff
@@ -521,6 +522,84 @@ TEST_F(CiderNewHashTableTest, aggUInt64LargeDatasetTest) {
   }
 }
 
+TEST_F(CiderNewHashTableTest, aggShortStringTest) {
+  using AggregatedHashTableForShortStringKey = StringHashMap<Block>;
+  AggregatedHashTableForShortStringKey ht_short_str;
+
+  std::vector<std::string> keys{"1", "2", "3", "4", "5"};
+  std::vector<int64_t> values{10, 20, 30, 40, 50};
+
+  for (int i = 0; i < keys.size(); i++) {
+    Block& block = ht_short_str[keys[i]];
+    block.add(values[i]);
+  }
+
+  for (int i = 0; i < keys.size(); i++) {
+    CHECK_EQ(ht_short_str[keys[i]].getSum(), values[i]);
+    CHECK_EQ(ht_short_str[keys[i]].getCount(), 1);
+  }
+
+  std::vector<std::string> keys2{"1", "2", "3", "4", "5"};
+  std::vector<int64_t> values2{50, 40, 30, 20, 10};
+  for (int i = 0; i < keys2.size(); i++) {
+    Block& block = ht_short_str[keys2[i]];
+    block.add(values2[i]);
+  }
+  for (int i = 0; i < keys.size(); i++) {
+    CHECK_EQ(ht_short_str[keys[i]].getSum(), 60);
+    CHECK_EQ(ht_short_str[keys[i]].getCount(), 2);
+    CHECK_EQ(ht_short_str[keys[i]].getAvg(), 30);
+  }
+}
+
+TEST_F(CiderNewHashTableTest, aggLongStringTest) {
+  using AggregatedHashTableForLongStringKey = HashMapWithSavedHash<StringRef, Block>;
```

Comment:
I think StrringRef usage is not safe. https://github.com/intel/BDTK/blob/main/cpp/src/cider/common/base/StringRef.h#L54

```
AggregatedHashTableForLongStringKey map;
{ 
std::string s = "a long string";
auto x = map[s];  // implicit conversion from std::string to StringRef, will record pointer of std::string
} // s deconstruct
auto x =map[s] // not safe!
```

Reply:
Yes, but `StringRef` is some kind a wrapper of `std::string`, which should have the same lifecycle. Besides, `std::string` is only used for test currently.

Reply:
My point is this API/usage is risky. you need to tell your user this limitation.